cst438week2_assignment2/src/main/java/cst438_assignment2/controllers/CityRestController.java

```diff
@@ -24,51 +28,16 @@
 	@Autowired
 	WeatherService weatherService;
 	
-//	@GetMapping("/cities/{name}")
-//	public String 
+	@Autowired
+	CityService cityService;
 	
 	@GetMapping("/api/cities/{name}")
-	public ResponseEntity<City> cityInfo(@PathVariable("name") String name ) {
+	public ResponseEntity<CityInfo> cityInfo(@PathVariable("name") String name ) {
+		CityInfo cityInfo = cityService.getCityInfo(name);
+				
+		return new ResponseEntity<CityInfo>(cityInfo, HttpStatus.OK);

```

Comment:
I have a CityInfo type class on my end, not a ResponseEntity... It works the way you did it! I would have to research more on what the difference is.